code/modules/research/nanites/nanite_programmer.dm

```diff
@@ -129,3 +130,9 @@
 				timer *= 10 //convert to deciseconds
 				program.timer_trigger_delay = timer
 			. = TRUE
+
+/obj/machinery/nanite_programmer/Hear(message, atom/movable/speaker, message_language, raw_message, radio_freq, list/spans, message_mode)
+	. = ..()
+	var/static/regex/when = regex("(?:^\\W*when|when\\W*$)", "i") //starts or ends with when
+	if(findtext(raw_message, when) && !istype(speaker, type))
```

Comment:
this line makes sure that two different nanite programmers won't bounce this phrase back and forth between themselves, right?

Reply:
actually in testing the same programmer would hear itself and recurse before runtiming on infinite loop, but it also covers that case too

Reply:
It wouldn't work if two different subtypes of `nanite_programmer` got together. Change `type` to `/obj/machinery/nanite_programmer` to fix that

Reply:
I could completely cover it by blacklisting the phrase "When you code it!!", after putting it in a define of course